modules/internal/ChapelBase.chpl

```diff
@@ -2375,16 +2375,23 @@ module ChapelBase {
   }
 
 
-  // This is a helper function that I injected to reduce the
-  // compiler's reliance on 'iterable.size' for coforall loops because
-  // we started generating warnings for the return type of
-  // '[range|domain|array].size' changing.
-  //
-  proc chpl_coforallSize(iterable) {
-    if (isRange(iterable) || isDomain(iterable) || isArray(iterable)) then
+  // Support for bounded coforall task counting optimizations
+  proc chpl_supportsBoundedCoforall(iterable, param zippered) param {
+    if zippered && isTuple(iterable) then
+      return chpl_supportsBoundedCoforall(iterable[0], zippered=false);
+    if isRange(iterable) || isDomain(iterable) || isArray(iterable) then
+      return true;
+    return false;
+  }
+
+  proc chpl_boundedCoforallSize(iterable, param zippered) {
```

Comment:
Did you drop the comment intentionally, or you don't think it is useful anymore? I think it pertains to the call to `sizeAs` but I am not sure.

Reply:
I dropped it intentionally. Now that there's more code than just the `.size` wrapper I didn't think we needed to justify why this is in module code.

Reply:
If it affects this conversation:  Once the .size warning goes away (next release I guess?), I was imagining we could/would take my helper function back out again to clean up the code.

Reply:
With the changes in this PR, I expect this support to stay in the module code, even if we switch to `.size`